### PR TITLE
Create Inputs documentation page

### LINKED ISSUE #3880: Attribute->Input rename - Documentation Update
### 🎛 Description

As part of #3802, update existing documentation to refer to attributes as inputs.

Retain mentions of the former name.

Check for broken links going to anchor URLs whose section names are changing.

### PR DESCRIPTION
**Pull request by clintoncwolfe**
## Description

Documents the Inputs system in detail, with new examples and coverage of the precedence system, clarification of how inheritance works, and differences in how the CLI provider works. Also coverage on how to debug inputs.

Also updates the glossary and profile page.

After merge, a website push will be needed.

## Related Issue

Fixes #3880

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -204,6 +200,16 @@ describe cars.where { engine_cylinders >= 6 } do
 end
 ```
 
+### input
+
+An _input_ is a value that Chef InSpec can source from a number of providers, including from the command line, profile metadata, or withing the control file DSL itself. You can use this feature either to change a [profile's](#profile) behavior by passing different attribute files or to store secrets that should not be directly present in a profile.
+
+Inputs were formerly known as `attributes`. Chef InSpec Inputs are unrelated to Chef attributes.
```
Comment (mjingle) [inline docs/glossary.md]: Would "Chef attributes" be Chef Infra attributes?

File: docs/inputs.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+As installed (without specialized plugins), Chef InSpec supports five ways of setting Inputs:
+
+ * Inline in control code, using `input('input_name', value: 42)`.
+ * In profile `inspec.yml` metadata files
+ * Using the CLI option `--input-file somefile.yaml`
+ * In kitchen-inspec, using the `verifier/inputs` settings
+ * In the Audit Cookbook, using the `node[:audit][:inputs]`
+
+In addition, Chef InSpec supports Input Plugins, which can provide optional integrations to specific key-value stores.
```
Comment (mjingle) [inline docs/inputs.md:78]: Is there a link to documentation for Input Plugins that can be included?

File: docs/inputs.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+dependent profile metadata < wrapper profile metadata
+
+This lets you override input values on the command line, as well as override child profile inline values from the parent profile.
+This matches the general behavior of InSpec v3, while also making some edge cases easier to reason about.
+
+#### The Details of Input Precedence
+
+Whenever an input provider sets a value on an input, a *priority value* is assigned to the operation.  Over the life of the input, multiple assignments, with varying priority values, may occur. When the input is evaluated, the current value is determined by finding the setting event with the highest priority.
+
+Note that this approach does not rely on execution order, nor does it rely on multiple named precedence levels. Each setting operation is preserved; this allows
```
Comment (mjingle) [inline docs/inputs.md]: The sentence at the end of this line needs to be completed.

File: docs/inputs.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+| CLI `--input-file` option              | 40       |   No                |
+| inspec-kitchen `inputs:` section       | 40       |   No                |
+| audit cookbook `node[:audit][:inputs]` | 40       |   No                |
+
+
+### What happened to "Attributes"?
+
+When originally introduced, the Input facility was named *Attributes*. This name was problematic, as the Chef Infra Client tool uses the same word to describe its parameterization system.
+
+ * Chef attributes have a completely different and much more complex precedence system
```
Comment (mjingle) [inline docs/inputs.md]: Why is there a bulleted list? Does a sentence need to be included above e.g. "Other reasons include:" ?

**Review by zenspider — APPROVED** [review]

File: docs/inputs.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+
+### Value
+
+Optional, any Ruby or YAML type. This is the value that will be available when you read the input. See [Reading Inputs](#reading-inputs-in-control-dsl).
+
+Allowed in: All
+
+### Type
+
+Optional, `String`, one of `String`, `Numeric`, `Regexp`, `Array`, `Hash`, `Boolean`, or `Any`. If provided, the value with be checked to see if it is of the corresponding type. Note that `Regexp` indicates that the value itself should be a regular expression, not that it should match any particular one.
```
Comment (mjingle) [inline docs/inputs.md]: The first section of information is a bit confusing. Can you please clarify about "Optional, `String`, one of `String`, `Numeric`, `Regexp`, `Array`, `Hash`, `Boolean`, or `Any`"? 

Is it that the type value is limited to `String`, `Numeric`, `Regexp`, `Array`, `Hash`, `Boolean`, or `Any`? More words may be needed to explain.

File: docs/inputs.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+
+All [input options](#input-options-reference) are supported in metadata files.
+
+There are two major advantages to defining inputs in profile metadata:
+ 1. The inputs and their configuration are listed explicitly in simple YAML in one place - a consumer of your profile does not need to read through the control code to find the inputs.
+ 2. You can set inputs in other profiles that you depend on using profile inheritance.
+
+### Using inputs with Profile inheritance
+
+When your profile relies on another profile using the `depends` key in the metadata file, you can set (that is, override) the value of the input in the dependent profile by including the `profile` option and naming the dependent profile.
```
Comment (mjingle) [inline docs/inputs.md]: I'm going to circle back to this sentence in my next pass. It needs to be clarified. For "When your profile relies on another profile using the `depends` key in the metadata file", is your profile relying on another profile using the `depends` key in the metadata file? Or is your profile using the `depends` key in the metadata file to rely on another profile? It's a bit ambiguous when first reading it.

**Review by miah — APPROVED** [review]

File: docs/inputs.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+dependent profile metadata < wrapper profile metadata
+
+This precedence lets you override input values on the command line, as well as override child profile inline values from the parent profile.
+This description matches the general behavior of InSpec v3, while also making some edge cases easier to reason.
+
+#### The Details of Input Precedence
+
+Whenever an input provider sets a value on an input, a *priority value* is assigned to the operation.  Over the life of the input, multiple assignments with varying priority values may occur. When the input is evaluated, the current value is determined by finding the setting event with the highest priority.
+
+Note that this approach does not rely on execution order, nor does it rely on multiple named precedence levels. Each setting operation is preserved; this allows the user to [debug](#debugging-inputs-with-the-event-log) the history of the input values.
```
Comment (mjingle) [inline docs/inputs.md]:
```suggestion
Note that this approach does not rely on execution order, nor does it rely on multiple named precedence levels. Each setting operation is preserved and this allows the user to [debug](#debugging-inputs-with-the-event-log) the history of the input values.
```

**Pull request merged**